### PR TITLE
[posix] add argument to set max power table

### PR DESCRIPTION
**Pull request by bukepo**
This commit adds a max power table for POSIX platform, which can customize the max allowed transmit power of each channel with the command line argument `--max-power-table`.

### CODE REVIEW

File: src/posix/platform/max_power_table.hpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+     * @params[in]  aChannel    The radio channel number.
+     * @params[in]  aPower      The max allowed transmit power in dBm.
+     *
+     */
+    void SetTransmitPower(uint8_t aChannel, int8_t aPower) { mPowerTable[aChannel - Radio::kChannelMin] = aPower; }
+
+    /**
+     * This method gets the allowed channel masks.
+     *
+     * All channels of max power value of 0xff is considered forbidden.
```
Comment (zhanglongxia) [inline src/posix/platform/max_power_table.hpp:64]: nit: 0x7f

File: src/posix/platform/radio.cpp [hunk 2]
```diff
@@ -50,6 +52,8 @@ static ot::Spinel::RadioSpinel<ot::Posix::SpiInterface, RadioProcessContext> sRa
 #error "OPENTHREAD_POSIX_CONFIG_RCP_BUS only allows OT_POSIX_RCP_BUS_UART and OT_POSIX_RCP_BUS_SPI!"
 #endif
 
+static ot::Posix::MaxPowerTable sMaxPowerTable;
```
Comment (zhanglongxia) [inline src/posix/platform/radio.cpp:55]: Do we need to move line 55 and 37 between `#if OPENTHREAD_POSIX_CONFIG_MAX_POWER_TABLE_ENABLE` and `#endif` ?

File: src/posix/platform/radio.cpp [hunk 3]
```diff
@@ -89,6 +93,22 @@ void otPlatRadioSetPromiscuous(otInstance *aInstance, bool aEnable)
 
 void platformRadioInit(const otPlatformConfig *aPlatformConfig)
 {
+#if OPENTHREAD_POSIX_CONFIG_MAX_POWER_TABLE_ENABLE
+    if (aPlatformConfig->mMaxPowerTable != NULL)
```
Comment (zhanglongxia) [inline src/posix/platform/radio.cpp:97]: How about moving the parsing string code to the class `MaxPowerTable` as a function?

File: script/test [hunk 4]
```diff
@@ -118,6 +118,7 @@ build_posix()
         "-DOT_PLATFORM=posix"
         "-DOT_READLINE=readline"
         "-DOT_THREAD_VERSION=${version}"
+        "-DOT_POSIX_MAX_POWER_TABLE=ON"
```
Comment (jwhui) [inline script/test:121]: nit: alphabetical ordering?

File: src/posix/main.c [hunk 5]
```diff
@@ -177,6 +182,13 @@ static void PrintUsage(const char *aProgramName, FILE *aStream, int aExitCode)
             "                                  MISO frame. Max value is 16.\n"
             "        --spi-small-packet=[n]    Specify the smallest packet we can receive in a single transaction.\n"
             "                                  (larger packets will require two transactions). Default value is 32.\n");
+#endif
+#if OPENTHREAD_POSIX_CONFIG_MAX_POWER_TABLE_ENABLE
+    fprintf(aStream,
+            "        --max-power-table         Max power for channels in ascendent order separated by commas,\n"
```
Comment (jwhui) [inline src/posix/main.c:188]:
```suggestion
            "        --max-power-table         Max power for channels in ascending order separated by commas,\n"
```

File: src/posix/main.c [hunk 6]
```diff
@@ -177,6 +182,13 @@ static void PrintUsage(const char *aProgramName, FILE *aStream, int aExitCode)
...
+            "                                  If the number of values is less than that of supported channels,\n"
+            "                                  the last value will be applied to all remaining channels.\n"
+            "                                  Special value 0x7f disable a channel.\n");
```
Comment (jwhui) [inline src/posix/main.c:191]:
```suggestion
            "                                  Special value 0x7f disables a channel.\n");
```

File: src/posix/platform/max_power_table.hpp [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+            {
+                channelMask |= (1 << i);
+            }
+        }
+
+        return channelMask;
+    }
+
+private:
+    static const int8_t kPowerForbidden = 0x7f;
```
Comment (jwhui) [inline src/posix/platform/max_power_table.hpp:85]: Why not enum, which provides greater flexibility in its use (e.g. switch statements, static variable initializer, etc.)?

File: src/posix/main.c [hunk 8]
```diff
@@ -177,6 +182,13 @@ static void PrintUsage(const char *aProgramName, FILE *aStream, int aExitCode)
             "                                  MISO frame. Max value is 16.\n"
             "        --spi-small-packet=[n]    Specify the smallest packet we can receive in a single transaction.\n"
             "                                  (larger packets will require two transactions). Default value is 32.\n");
+#endif
+#if OPENTHREAD_POSIX_CONFIG_MAX_POWER_TABLE_ENABLE
+    fprintf(aStream,
+            "        --max-power-table         Max power for channels in ascending order separated by commas,\n"
+            "                                  If the number of values is less than that of supported channels,\n"
+            "                                  the last value will be applied to all remaining channels.\n"
+            "                                  Special value 0x7f disables a channel.\n");
```
Comment (abtink) [inline src/posix/main.c:191]:
```suggestion
            "                                  Special value 0x7f disallows a channel.\n");
```

**Pull request merged**